book/src/development/method_checks.md

```diff
@@ -0,0 +1,92 @@
+# Method Checks
+
+In some scenarios we might want to check for methods when developing
+a lint. There are two kinds of questions that we might be curious about:
+
+- Invocation: Does an expression call a specific method?
+- Definition: Does the type `Ty` of an expression define a method?
+
+## Checking if an expr is calling a specific method
+
+Suppose we have an `expr`, we can check whether it calls a specific
+method, e.g. `our_fancy_method`, by performing a pattern match on
+the [ExprKind] that we can access from `expr.kind`:
+
+```rust
+use rustc_hir as hir;
+use rustc_lint::{LateContext, LateLintPass};
+use rustc_span::sym;
+
+impl<'tcx> LateLintPass<'tcx> for OurFancyMethodLint {
+    fn check_expr(&mut self, cx: &LateContext<'tcx>, expr: &'tcx hir::Expr<'_>) {
+        // Check our expr is calling a method with pattern matching
+        if let hir::ExprKind::MethodCall(path, _, [_self_arg, ..]) = &expr.kind
+            // Check the name of this method is `some_method`
+            && path.ident.name == sym!(our_fancy_method)
+            // Optionally, we can check the type of the self argument whenever necessary.
+            // See "Type Checking" chapter of the Clippy book for more information.
+        {
+                println!("`expr` is a method call for `our_fancy_method`");
+        }
+    }
+}
+```
+
+Take a closer look at the `ExprKind` enum variant [MethodCall] for more
+information on the pattern matching.
+As mentioned in [Define Lints](define_lints.md#lint-groups),
+the `methods` module is full of pattern matching with `Methodcall`
+in case the reader wishes to explore more.
+
+Additionally, we use Clippy utils [sym] macro to conveniently compare
+an input `our_fancy_method` into a `Symbol` since the `ident` [Ident]
+in [PathSegment], which is a value of the `MethodCall` enum variant,
+contains a field `name` that is defined as a `Symbol`.
+
+## Checking if a type defines a specific method
```

Comment:
You could also add a reference to the section about checking if a type implements a certain trait. I can imagine this part could also be confused with that.

Otherwise, this is solid and a fun read. I'm looking forward to merging this :)